### DISCUSSION TITLE
Unable to implement user login  Authentik API

### DISCUSSION THREAD

**Discussion by kiran-bal** [goauthentik/authentik#7228] [Q&A]
I'm currently working on integrating user authentication into my application using the Authentik API. I've successfully added a user through the core/user endpoint and set a password. However, I'm having trouble finding the right API path for user login in the Authentik API browser. It seems there's no obvious endpoint for user authentication.

please provide some insights into the correct API endpoint or method for user login. I need to authenticate users using their usernames and passwords and obtain a token for subsequent API requests

**Comment by BeryJu** [comment]
See https://goauthentik.io/developer-docs/api/flow-executor for a short introduction

**Reply by kiran-bal** [reply]
Thank you for the assistance. I've gone through the documentation, and it's been quite informative. However, I came across the term "component" in the context of "ak-stage-identification," and I'd like to understand this better.

Could you please provide some clarification on what "component" means in this context and how it's defined? 
also, how can I use the created password for the authentication process?

**Reply by BeryJu** [reply]
The component attribute is mainly for the web-based executor to know which stage interface to render, but it's a unique identifier for which stage is currently active (the API schema shows a list of all of the options)

You can provide the password in the second request after submitting the username/email, or you can configure the identification stage to also have a password stage, and with that you can do both in one single request

**Reply by kiran-bal** [reply]
Thank you so much for your response! Your explanation about the component attribute and the authentication process was very helpful. I appreciate your time and expertise in clarifying these details. If I have any more questions, I'll reach out. Thanks again!

**Reply by BahaKhmeissi** [reply]
Hello , what about sendnig the confirmation email after setting the password ?  how can i execute that stage through an api ?
